includes/multi-currency/Helpers/OrderMetaHelper.php

```diff
@@ -0,0 +1,443 @@
+ */
+
+namespace WCPay\MultiCurrency\Helpers;
+
+use WC_Payments_API_Client;
+use WC_Payments_Features;
+use WC_Payments_Utils;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that controls OrderMetaHelper functionality.
+ */
+class OrderMetaHelper {
+	/**
+	 * Client for making requests to the WooCommerce Payments API.
+	 *
+	 * @var WC_Payments_API_Client
+	 */
+	private $payments_api_client;
+
+	/**
+	 * Array of errors, if any.
+	 *
+	 * @var array
+	 */
+	private $errors = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Payments_API_Client $payments_api_client Payments API client.
+	 */
+	public function __construct( WC_Payments_API_Client $payments_api_client ) {
+		$this->payments_api_client = $payments_api_client;
+		$this->add_actions();
+	}
+
+	/**
+	 * Adds the button to toggle the tool on and off under WooCommerce > Status > Tools.
+	 *
+	 * @param array $tools The current array of tools.
+	 *
+	 * @return array The modified tools array.
+	 */
+	public function add_tool_button( $tools ) {
+		/**
+		 * We are not able to look at the current status of the tool to dynamically change these labels. The tool runs after the labels are created.
+		 */
+		$tools['wcpay_mc_meta_helper_toggle'] = [
+			'name'     => __( 'Enable/Disable Multi-Currency Meta Helper', 'woocommerce-payments' ),
+			'button'   => __( 'Enable/Disable', 'woocommerce-payments' ),
+			'desc'     => __( 'This tool can be used to fix meta data on orders related to Multi-Currency conversion rates.', 'woocommerce-payments' ),
+			'callback' => [ $this, 'toggle_helper_option' ],
+		];
+
+		return $tools;
+	}
+
+	/**
+	 * Toggles the feature on and off via database option.
+	 *
+	 * @return string Whether the feature has been enabled or disabled.
+	 */
+	public function toggle_helper_option() {
+		if ( WC_Payments_Features::is_mc_order_meta_helper_enabled() ) {
+			update_option( WC_Payments_Features::MC_ORDER_META_HELPER_FLAG_NAME, 0 );
+			return __( 'Multi-Currency Meta Helper has been disabled', 'woocommerce-payments' );
+		}
+
+		update_option( WC_Payments_Features::MC_ORDER_META_HELPER_FLAG_NAME, 1 );
+		return __( 'Multi-Currency Meta Helper has been enabled', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Outputs and then clears our errors if there are any.
+	 *
+	 * @return void
+	 */
+	public function maybe_output_errors() {
+		if ( $this->has_errors() ) {
+			foreach ( $this->errors as $error ) {
+
+				?>
+				<div class="error notice">
+					<p><strong>Error: </strong><?php echo esc_html( $error ); ?></p>
+				</div>
+				<?php
+			}
+			$this->clear_errors();
+		}
+	}
+
+	/**
+	 * Updates the exchange rate meta data.
+	 *
+	 * @param int $order_id The order we are working with.
+	 *
+	 * @return void
+	 */
+	public function maybe_update_exchange_rate( $order_id ) {
+		// Verify nonce.
+		$nonce_value = ! empty( $_POST['wcpay_multi_currency_exchange_rate_nonce'] ) ? sanitize_text_field( wp_unslash( $_POST['wcpay_multi_currency_exchange_rate_nonce'] ) ) : false;
```

Comment:
Maybe we should also add a role/permission check here? To prevent unauthorized users to change rates?

Reply:
Good catch on that. I didn't think to add it because the form and actions only work inside of the order page editing context, but better safe than sorry. Added in: https://github.com/Automattic/woocommerce-payments/pull/6582/commits/710beecd504906ed5ff896a55012d58cda775092